lib/driver.js

```diff
@@ -412,10 +412,20 @@ class XCUITestDriver extends BaseDriver {
       await this.wda.uninstall();
       this.logEvent('wdaUninstalled');
     } else if (!util.hasValue(this.wda.webDriverAgentUrl) && (await this.wda.isRunning())) {
-      log.info(`Will reuse previously cached WDA instance at '${this.wda.url.href}'. ` +
-               `Set the wdaLocalPort capability to a value different from ${this.wda.url.port} ` +
-               `if this is an undesired behavior.`);
-      this.wda.webDriverAgentUrl = this.wda.url.href;
+      let currentAppBundleID;
+      if (util.hasValue(this.opts.updatedWDABundleId)) {
+        let response = await this.proxyCommand('/status', 'GET');
+        currentAppBundleID = response.currentApp.bundleID;
+      }
+      if (this.opts.updatedWDABundleId !== currentAppBundleID) {
```

Comment:
it makes sense to check this only if `this.opts.updatedWDABundleId` is set, so I'd rather add ` && util.hasValue(currentAppBundleID)` check